tests/lib/Token/WebSafeGeneratorTest.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Tests\Core\Token;
+
+use Ibexa\Contracts\Core\Token\TokenGeneratorInterface;
+use Ibexa\Core\Token\WebSafeGenerator;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @covers \Ibexa\Core\Token\WebSafeGenerator
+ */
+final class WebSafeGeneratorTest extends TestCase
+{
+    private TokenGeneratorInterface $tokenGenerator;
+
+    /** @var \Ibexa\Contracts\Core\Token\TokenGeneratorInterface&\PHPUnit\Framework\MockObject\MockObject */
+    private TokenGeneratorInterface $randomBytesTokenGenerator;
+
+    protected function setUp(): void
+    {
+        $this->randomBytesTokenGenerator = $this->createMock(TokenGeneratorInterface::class);
+        $this->tokenGenerator = new WebSafeGenerator($this->randomBytesTokenGenerator);
+    }
+
+    /**
+     * @dataProvider provideDataForTestGenerateToken
+     *
+     * @throws \Exception
+     */
+    public function testGenerateToken(
+        int $expectedTokenLength,
+        string $mockGeneratorOutputToken,
+        string $expectedToken
+    ): void {
+        $this->mockTokenGeneratorGenerateToken(
+            strlen($mockGeneratorOutputToken),
```

Comment:
We expect the `WebSafeGenerator` to "pass through" the argument value - so it's more accurate to pass that.
```suggestion
            $expectedTokenLength,
```

Because we expect token generators to return exactly that amount of characters, I do think that testing against shorter/longer outputs from internal generator is not necessary (just noting that, because it crossed my mind).